src/app/services/scheduler/scheduler.service.ts

```diff
@@ -325,48 +325,66 @@ export class SchedulerService {
   }
 
   private formatResource(resource: SchedulerResourceInfo): string {
-    const formatted = [];
-
-    if (resource && resource.memory !== undefined) {
-      formatted.push(`Memory: ${CommonUtil.formatBytes(resource.memory)}`);
-    } else {
-      formatted.push(`Memory: ${NOT_AVAILABLE}`);
-    }
-
-    if (resource && resource.vcore !== undefined) {
-      formatted.push(`CPU: ${CommonUtil.formatCount(resource.vcore)}`);
-    } else {
-      formatted.push(`CPU: ${NOT_AVAILABLE}`);
-    }
+    const formatted: string[] = [];
+    if (resource) {
+      // Object.keys() didn't guarantee the order of keys, sort it before iterate.
+      Object.keys(resource).sort(this.resourcesCompareFn).forEach((key) => {
+        let value = resource[key];
+        let formattedKey = key;
+        let formattedValue = NOT_AVAILABLE;
 
-    if (resource){
-      Object.keys(resource).forEach((key) => {
         switch(key){
           case "memory":
-          case "vcore":{
+            formattedKey = "Memory";
+            formattedValue = CommonUtil.formatMemoryBytes(value);
+            break;
+          case "vcore":
+            formattedKey = "CPU";
+            formattedValue = CommonUtil.formatCpuCore(value);
             break;
-          }
-          case "ephemeral-storage":{
-            if (resource[`ephemeral-storage`] == 0) {
-              formatted.push(`ephemeral-storage: ${NOT_AVAILABLE}`);
-            }else{
-              formatted.push(`ephemeral-storage: ${CommonUtil.formatBytes(resource[key])}`);
+          case "ephemeral-storage":
+            if (value !== 0){  // if value is 0, show NOT_AVAILABLE
+              formattedValue = CommonUtil.formatEphemeralStorageBytes(value);
             }
             break;
-          }
-          default:{
-            if (resource[key] == 0) {
-              formatted.push(`${key}: ${NOT_AVAILABLE}`);
-            }else{
-              formatted.push(`${key}: ${CommonUtil.formatOtherResource(resource[key])}`);
+          default:
+            if (value !== 0){ // if value is 0, show NOT_AVAILABLE
+              if (key.startsWith('hugepages-')) {
+                formattedValue = CommonUtil.formatMemoryBytes(value);
+              } else{
+                formattedValue = CommonUtil.formatOtherResource(value);
+              }
             }
             break;
-          }
-        }
+         }
+        // }
+        formatted.push(`${formattedKey}: ${formattedValue}`);
       });
     }
-    
+
+    // Set CPU/Memory as NOT_AVAILABLE if it was not in Object.keys() or resource is undefined.
+    if (!resource || resource.vcore === undefined)
+      formatted.unshift(`CPU: ${NOT_AVAILABLE}`);
+    if (!resource || resource.memory === undefined)
+      formatted.unshift(`Memory: ${NOT_AVAILABLE}`);
```

Comment:
See comment added I don't think we should do this.

Reply:
Removed.